content/blog/kubernetes-yaml-generation/index.md

```diff
@@ -0,0 +1,208 @@
+
+Here's what the resulting directory looks like:
+
+![Rendered Manifests](render-directory.png)
+
+Note that `CustomResourceDefinitions` need to be applied first, so they are rendered in a separate subdirectory. You
+could deploy these with `kubectl` like this:
+
+```shell script
+kubectl apply -f "${RENDER_DIRECTORY}/0-crd"
+kubectl apply -f "${RENDER_DIRECTORY}/1-manifest"
+```
+
+Voilà! From 33 (44 with whitespace and comments) lines of TypeScript code to 102 lines of YAML!
+
+Here's the rendered `Deployment` resource.
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  annotations:
+    foo: bar
+    kubectl.kubernetes.io/last-applied-configuration: |
+      {"apiVersion":"apps/v1","kind":"Deployment","metadata":{"annotations":{"foo":"bar","pulumi.com/autonamed":"true"},"labels":{"app.kubernetes.io/managed-by":"pulumi"},"name":"nginx-nyn4tlkx"},"spec":{"replicas":3,"selector":{"matchLabels":{"app":"nginx"}},"template":{"metadata":{"labels":{"app":"nginx"}},"spec":{"containers":[{"env":[{"name":"CONFIG","valueFrom":{"configMapKeyRef":{"key":"config","name":"cm-5opqxhna"}}},{"name":"PASSWORD","valueFrom":{"secretKeyRef":{"key":"password","name":"secret-hfcg0l06"}}}],"image":"nginx","name":"nginx","ports":[{"containerPort":8080,"name":"http"}],"volumeMounts":[{"mountPath":"/data","name":"data-4qulussd"}]}],"volumes":[{"name":"data-4qulussd","persistentVolumeClaim":{"claimName":"data-4qulussd"}}]}}}}
+    pulumi.com/autonamed: "true"
+  labels:
+    app.kubernetes.io/managed-by: pulumi
+  name: nginx-nyn4tlkx
+spec:
+  replicas: 3
+  selector:
+    matchLabels:
+      app: nginx
+  template:
+    metadata:
+      labels:
+        app: nginx
+    spec:
+      containers:
+      - env:
+        - name: CONFIG
+          valueFrom:
+            configMapKeyRef:
+              key: config
+              name: cm-5opqxhna
+        - name: PASSWORD
+          valueFrom:
+            secretKeyRef:
+              key: password
+              name: secret-hfcg0l06
+        image: nginx
+        name: nginx
+        ports:
+        - containerPort: 8080
+          name: http
+        volumeMounts:
+        - mountPath: /data
+          name: data-4qulussd
+      volumes:
+      - name: data-4qulussd
+        persistentVolumeClaim:
+          claimName: data-4qulussd
+```
+
+See [this gist](https://gist.github.com/lblackstone/686935edf7fdcd23d916f34d35bba64a) for the complete rendered output.
+
+## Caveats
+
+There are two important caveats to note about YAML rendering support:
+
+1. The YAML-rendered resources are not created on a Kubernetes cluster, so information that is computed server-side
+will not be available in your program. For example, a `Service` will not have IP assignments, so attempting to export
+these values will not work as usual (the value will be `undefined`).
+1. **Any Secret values will appear in plaintext in the rendered manifests.** This includes any values marked as
```

Comment:
Prefix with "Warning:" or some such - and maybe a tracking issue if one exists?

Reply:
We don't have current plans for changing this, so there's no tracking issue. I'm planning to wait on user feedback to figure out how to proceed.